### PR TITLE
Ros2 refactor continued 

### PR DESCRIPTION
**Pull request by SteveMacenski**
- Using smart pointers throughout
- Moved "main" files and capabilities into src directory with cpp implementations
- Moved Critics to a subdirectory off of include to match what already existed in src
- Fixed some more includes for ROS 2 styling
- Changed private to protected so overriding classes can use (and changed more final's to override's)
- Changing all filenames to use snake_case to align with standard

### CODE REVIEW

File: include/mppic/trajectory_visualizer.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  TrajectoryVisualizer() = default;
+
+  void on_configure(rclcpp_lifecycle::LifecycleNode::WeakPtr parent, const std::string & frame_id);
+  void on_cleanup();
+  void on_activate();
+  void on_deactivate();
+
+  void reset();
+
+  template <typename Container> void add(Container && trajectory);
```
Comment (artofnothingness) [inline include/mppic/trajectory_visualizer.hpp:24]: Definition must be here, otherwise it wouldn't work out. Other approach is to create impl hpp file

**Review by artofnothingness — CHANGES_REQUESTED** [review]

**Comment by artofnothingness** [comment]
Looks nice! Could you try to run optimizer_test to see if it works ?

File: include/mppic/trajectory_visualizer.hpp (hunk 1 above)
Comment (artofnothingness) [reply include/mppic/trajectory_visualizer.hpp:24]: Merged. i'll try to fix it if it breaks.

**Comment by SteveMacenski** [comment]
Whoops, sorry, I've been working on this in the late afternoons after I get my overhead activities done (PR reviews, issue triage, emails, etc) so I hadn't gotten to your comments yet.

**Comment by artofnothingness** [comment]
It's ok) I just thought I could fix it quickly)

**Pull request merged**